tstest/integration/integration_test.go

```diff
@@ -311,6 +311,96 @@ func TestAddPingRequest(t *testing.T) {
 	t.Error("all ping attempts failed")
 }
 
+func TestTaildrop(t *testing.T) {
+	t.Parallel()
+	bins := BuildTestBinaries(t)
+
+	env := newTestEnv(t, bins, configureControl(func(control *testcontrol.Server) {
+		control.AllNodesSameUser = true
+	}))
+	defer env.Close()
+
+	n1 := newTestNode(t, env)
+	n1SocksAddrCh := n1.socks5AddrChan()
+	d1 := n1.StartDaemon(t)
+	defer d1.Kill()
+
+	n2 := newTestNode(t, env)
+	n2SocksAddrCh := n2.socks5AddrChan()
+	d2 := n2.StartDaemon(t)
+	defer d2.Kill()
+
+	n1Socks := n1.AwaitSocksAddr(t, n1SocksAddrCh)
+	n2Socks := n1.AwaitSocksAddr(t, n2SocksAddrCh)
+	t.Logf("node1 SOCKS5 addr: %v", n1Socks)
+	t.Logf("node2 SOCKS5 addr: %v", n2Socks)
+
+	for _, n := range env.Control.AllNodes() {
+		n.Capabilities = append(n.Capabilities, tailcfg.CapabilityFileSharing)
+	}
+
+	n1.AwaitListening(t)
+	n2.AwaitListening(t)
+	n1.MustUp()
+	n2.MustUp()
+	n1.AwaitRunning(t)
+	n2.AwaitRunning(t)
+
+	target := n2.AwaitIP(t)
+
+	tempDir := t.TempDir()
+	tmpFile, err := ioutil.TempFile(tempDir, "taildrop")
```

Comment:
seems overkill to use TempFile in an empty directory.

just use ioutil.WriteFile?